test/functionalTest/cases/3068_non_legacy_notifications/non_legacy_notifications_with_providers.test

```diff
@@ -0,0 +1,395 @@
+# Copyright 2019 Telefonica Investigacion y Desarrollo, S.A.U
+#
+# This file is part of Orion Context Broker.
+#
+# Orion Context Broker is free software: you can redistribute it and/or
+# modify it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# Orion Context Broker is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero
+# General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with Orion Context Broker. If not, see http://www.gnu.org/licenses/.
+#
+# For those usages not covered by this license please contact with
+# iot_support at tid dot es
+
+# VALGRIND_READY - to mark the test ready for valgrindTestSuite.sh
+
+--NAME--
+Query forwarding in NGSIv2
+
+--SHELL-INIT--
+dbInit CB
+dbInit CP1
+dbInit CP2
+dbInit CP3
+dbInit CP4
+brokerStart CB 38,186-187,235 IPV4
+brokerStart CP1
+brokerStart CP2
+brokerStart CP3
+brokerStart CP4
+
+--SHELL--
+
+#
+# 01. Register entities of type T, with ID .*, and attribute A1, for CP1
+# 02. Register entities of type T, with ID .*, and attribute A2, for CP2
+# 03. Register entities of type T, with ID .*, and attribute A3, for CP3
+# 04. Register entities of type U, with ID .*, and attribute A4, for CP4
+# 05. Create E1/T in CP1
+# 06. Create E2/T in CP2
+# 07. Create E3/T in CP3
+# 08. Create E4/U in CP4
+# 09. Query entities of type T - see E1, E2 and E3
+# 10. Query entities of type U - see E4
+# 11. Query entities of type T and with attr A1 - see E1 with A1, and E2+E3 without attrs
```

Comment:
This is a bit weird. I'd say that the right one should be:

* Query entities of type T and with attr A1 - see E1 with A1

However, if the behaviour is the same for NGSIv1 based registration, it would be out of this PR so ok (the NGSIv2 behaviour shoul "mimic" the NGSIv1, issues included :). Could you check and tell, pls?

Reply:
All matching entities (not looking at attributes) are returned.
If they have any attributes among the ones in the URI param `attrs`, then those attributes are included.
If no attributes among the ones in the URI param `attrs`, then the entities enter "naked" - only entity ID and Type.
Seems like V1 works like this.

Reply:
Let's transform that "it seems like" in "it works that way" to be fully sure :)

Reply:
Bad choice of words  (_seems like V1 works like this_) ...
What is done is that all entities that match (using entity id and type) are included.
After that, the attributes to be included are filtered according to the attr-list.
Entities that have no attributes of that list are returned without attributes.

NTC

Reply:
To check if NGSIv1 behaves the same.

Reply:
I suggest to add a FIXME in the .test preamble with this (or a similar) wording:

> FIXME: step 11 should be just "Query entities of type T and with attr A1 - see E1 with A1", I mean without "and E2+E3 without attrs". To do so, CB should be more precise in the way it invokes CPrs or in the way it filters the CPr results before responding to the client. Note this is not a NGSIv2-specific flaw, NGSIv1 forwarding behaves the same way.

Reply:
Fixed in aef7cb8